test/extended/jenkins/plugin.go

```diff
@@ -1,105 +1,279 @@
 package jenkins
 
 import (
-	g "github.com/onsi/ginkgo"
-	o "github.com/onsi/gomega"
-
 	"bytes"
+	"encoding/xml"
 	"fmt"
 	"io"
 	"io/ioutil"
 	"net/http"
 	"regexp"
+	"strings"
 	"time"
 
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	kapi "k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/util/wait"
 
 	exutil "github.com/openshift/origin/test/extended/util"
 )
 
-func immediateInteractionWithJenkins(uri, method string, body io.Reader, status int) {
-	req, err := http.NewRequest(method, uri, body)
-	o.Expect(err).NotTo(o.HaveOccurred())
+type JenkinsRef struct {
+	oc   *exutil.CLI
+	host string
+	port string
+}
+
+// Struct can be marshalled into XML to represent a Jenkins workflow job definition.
+type FlowDefinition struct {
+	XMLName          xml.Name `xml:"flow-definition"`
+	Plugin           string   `xml:"plugin,attr"`
+	KeepDependencies bool     `xml:"keepDependencies"`
+	Definition       Definition
+}
+
+type Definition struct {
+	XMLName xml.Name `xml:"definition"`
+	Class   string   `xml:"class,attr"`
+	Plugin  string   `xml:"plugin,attr"`
+	Script  string   `xml:"script"`
+}
+
+// Builds a URI for the Jenkins server.
+func (j *JenkinsRef) buildURI(resourcePathFormat string, a ...interface{}) string {
+	resourcePath := fmt.Sprintf(resourcePathFormat, a...)
+	return fmt.Sprintf("http://%s:%v/%s", j.host, j.port, resourcePath)
+}
+
+// Submits a GET request to this Jenkins server.
+// Returns a response body and status code or an error.
+func (j *JenkinsRef) getResource(resourcePathFormat string, a ...interface{}) (string, int, error) {
+	uri := j.buildURI(resourcePathFormat, a...)
+	req, err := http.NewRequest("GET", uri, nil)
+	if err != nil {
+		return "", 0, fmt.Errorf("Unable to build request for uri %q: %v", uri, err)
+	}
+
+	req.SetBasicAuth("admin", "password")
```

Comment:
heads up ... assuming my PR ever gets through the merge queue, you'll need to rebase and plug in my retrieval of the password for the jenkins admin user here.